hack/update-go-module-dependencies.sh

```diff
@@ -57,10 +65,23 @@ for x in vendor/github.com/openshift/*; do
 	esac
 done
 
-for x in aws/v2 azure openstack; do
-	go mod edit -replace sigs.k8s.io/cluster-api-provider-$x="$(go list -mod=mod -m github.com/openshift/cluster-api-provider-$x@$RELEASE | sed -e 's/ /@/')"
+for x in vendor/k8s.io/*; do
+  case $x in
+    # skip, it's replaced by openshift
+    vendor/k8s.io/cloud-provider-vsphere)
+      ;;
+    *)
+      # ignore errors, they are from components which aren't tagged following the v0.x.y schema
+      go mod edit -replace "${x##vendor/}"="$(go list -mod=mod -m ${x##vendor/}@$K8S_RELEASE | sed -e 's/ /@/')" || true
```

Comment:
Could this cause problems for components that despite having a semver, are not aligned with the K8s release version? (e.g., if at some point they happened to have an older tag with the same number as K8s release version, but not being part of that release)

Reply:
Fair point. However, I think it would be a great coincidence if this happens, and it should be catched when reviewing the actual changes in go.mod. Not sure if it's worth maintaining the list of dependencies which follow the the k8s release schema (or those who don't) 🤔 Let me check how long that list would be...

Reply:
Fixed